ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenRunner.java

```diff
@@ -185,10 +188,29 @@ public Void visitFunctionCall(final FunctionCall node, final TypeContext context
       final List<SqlArgument> argumentTypes = new ArrayList<>();
       final FunctionName functionName = node.getName();
       for (final Expression argExpr : node.getArguments()) {
-        process(argExpr, null);
-        argumentTypes.add(SqlArgument.of(
-            expressionTypeManager.getExpressionSqlType(argExpr)
-        ));
+        process(argExpr, context);
+        final SqlType newSqlType = expressionTypeManager.getExpressionSqlType(argExpr, context);
+        // for lambdas - if we see this it's the  array/map being passed in we save the type
+        if (context.notAllInputsSeen()) {
```

Comment:
not part of this PR, but I think I might be missing something 
```java
  public boolean notAllInputsSeen() {
    return lambdaInputTypeMapping.size() != lambdaInputTypes.size() || lambdaInputTypes.size() == 0;
  }
```

Can this handle lambdas with 0 arguments properly? If not that can be OK but we should make sure we fail that if anyone tries to register one.

Reply:
Hmm... I think it would register more input types than we want it to, but I think lambdas with 0 arguments would still fail. Would that be something like `array_transform(testval1, 5 => 5)`? or `map_transform(MAP(5:=3, 1:=1), () => 2, () => 5)`?

Reply:
Just looked back into it, in `LambdaFunctionCall` we check the argument size
`    if (arguments.size() == 0) {
      throw new IllegalArgumentException(
          String.format("Lambda expression must have at least 1 argument. => %s", body.toString()));
    }`